django/views/generic/edit.py

```diff
@@ -112,7 +112,15 @@ def form_invalid(self, form):
         If the form is invalid, re-render the context data with the
         data-filled form and errors.
         """
-        return self.render_to_response(self.get_context_data(form=form))
+        return self.render_to_response(self.get_context_data())
+
+    def get_context_data(self, **kwargs):
+        """
+        Insert the form into the context dict.
+        """
+        context = {'form': self.get_form()}
+        context.update(kwargs)
```

Comment:
For some reason neither ContextMixin nor SingleObjectMixin use setdefault, so we can change it just here, change it in these 3 places, or leave as is. What's the general django policy on use of setdefault?